spec/datadog/core/workers/runtime_metrics_spec.rb

```diff
@@ -282,11 +282,11 @@
       it 'produces metrics every interval' do
         worker.perform
 
-        sleep(default_flush_interval * 1.1)
+        sleep(default_flush_interval * 2)
```

Comment:
Could we try not using `sleep`

```ruby
describe 'interval' do
  let(:default_flush_interval) { 0.1 }

  before do
    stub_const(
      'Datadog::Core::Workers::RuntimeMetrics::DEFAULT_FLUSH_INTERVAL',
      default_flush_interval
    )
  end

  after { worker.stop }

  let(:metrics) do
    class StubMetrics
      attr_reader :flush_count

      def initialize
        @flush_count = 0
      end

      def flush
        @flush_count+=1
      end
    end.new
  end

  it 'produces metrics every interval' do
    worker.perform

    begin
      Timeout::timeout(5) {
        loop do
          break if metrics.flush_count >= 2
        end
      }
      expect(true).to eq(true)
    rescue Timeout::Error
      fail 'metrics are not flush every interval'
    end
  end
end
```

Reply:
I know timeout is a sort of sleep in a way, but I feel that it less magical 😄

Reply:
And yeah, timeout is indeed basically a sleep (although they use wait on a conditional variable): https://github.com/ruby/ruby/blob/1a4b4cd7f8b53fc52191c1ddcf3f624a5e687c24/lib/timeout.rb#L113 .

Reply:
If we have a dedicated method for doing this sort of things I'm all in favour of using it. Even if under the hood it uses sleep 😉 

I suggested `Timeout` because as you stated it uses a conditional variable which in my opinion is better use case for this kind of scenarios

Reply:
I would consider any usage of `Timeout` _very carefully_: https://jvns.ca/blog/2015/11/27/why-rubys-timeout-is-dangerous-and-thread-dot-raise-is-terrifying/

Reply:
I agree that waiting for the actual condition is more correct here. I'll change the condition.